spotify_tensorflow/luigi/python_dataflow_task.py

```diff
@@ -0,0 +1,303 @@
+                names = ["%s-%s" % (name, key) for key in targets.keys()]
+            else:
+                names = [name] * len(uris)
+            for (arg_name, uri) in zip(names, uris):
+                input_args.append("--%s=%s" % (arg_name, uri))
+
+        return input_args
+
+    def _get_output_args(self):
+        if not isinstance(self._output, dict):
+            raise ValueError("Output must be dict type")
+
+        output_args = []
+        self._output_uris = {}
+
+        for (name, target) in self._output.items():
+            uri = target.generate_uri() if hasattr(target, "generate_uri") else get_uri(target)
+            uri = uri.rstrip("/")
+            output_args.append("--%s=%s" % (name, uri))
+            self._output_uris[name] = uri
+
+        return output_args
+
+    def _get_runner(self):
+        return "DirectRunner" if self.local_runner else "DataflowRunner"
+
+    def _get_dataflow_args(self):
+        dataflow_args = []
+
+        _runner = self._get_runner()
+        if _runner:
+            dataflow_args += ["--runner={}".format(_runner)]
+        if self.project:
+            dataflow_args += ["--project={}".format(self.project)]
+        if self.staging_location:
+            dataflow_args += ["--staging_location={}".format(self.staging_location)]
+        if self.zone:
+            dataflow_args += ["--zone={}".format(self.zone)]
+        if self.region:
+            dataflow_args += ["--region={}".format(self.region)]
+        if self.temp_location:
+            dataflow_args += ["--temp_location={}".format(self.temp_location)]
+        if self.num_workers:
+            dataflow_args += ["--num_workers={}".format(self.num_workers)]
+        if self.autoscaling_algorithm:
+            dataflow_args += ["--autoscaling_algorithm={}".format(self.autoscaling_algorithm)]
+        if self.max_num_workers:
+            dataflow_args += ["--max_num_workers={}".format(self.max_num_workers)]
+        if self.network:
+            dataflow_args += ["--network={}".format(self.network)]
+        if self.subnetwork:
+            dataflow_args += ["--subnetwork={}".format(self.subnetwork)]
+        if self.disk_size_gb:
+            dataflow_args += ["--disk_size_gb={}".format(self.disk_size_gb)]
+        if self.worker_machine_type:
+            dataflow_args += ["--worker_machine_type={}".format(self.worker_machine_type)]
+        if self.job_name:
+            dataflow_args += ["--job_name={}".format(self.job_name)]
+        if self.worker_disk_type:
+            dataflow_args += ["--worker_disk_type={}".format(self.worker_disk_type)]
+        if self.service_account:
+            dataflow_args += ["--service_account_email={}".format(self.service_account)]
+        if self.requirements_file:
+            dataflow_args += ["--requirements_file={}".format(self.requirements_file)]
+
+        return dataflow_args
+
+    def tfx_args(self):
```

Comment:
should we couple a generic python dataflow task with tfx specific one?

Reply:
Good point. The reason I included it here is to have a placeholder for individual tfx component to override it for its own args. Maybe it's over ambitious to name the class as `PythonDataflowTask`, since it's only served as the base class for tfx jobs.

Reply:
+1 with decoupling this - my personal preference would be to keep this one generic (since it could be re-used as a generic python dataflow task, and introduce a thin, intermediate `TfxBaseTask` that extends this with `def tfxArgs`, and overrides `_mk_cmd_line`. Thoughts?

Reply:
+1

Reply:
sounds good to me!